astropy/cosmology/core.py

```diff
@@ -3309,3 +3357,11 @@ def inf_like(x):
         return np.inf
     else:
         return np.full_like(x, np.inf, dtype='float')
+
+
+# modified from https://stackoverflow.com/a/33607093
+def _get_subclasses(cls):
```

Comment:
Maybe instead make this a class method on Cosmology? Or is this generically useful in astropy/utils ?

Reply:
Did the former. Looks much better.